crates/solidity/testing/snapshots/cst_output/Expression/function_call_options/generated/0.6.2-success.yml

```diff
@@ -15,27 +15,30 @@ Tree:
-                  - OpenBrace (Token): "{" # 3..4
-                  - NamedArgumentsList (Rule): # 4..20 "value: 0, gas: 1"
-                      - NamedArgument (Rule): # 4..12 "value: 0"
-                          - Identifier (Token): "value" # 4..9
-                          - Colon (Token): ":" # 9..10
-                          - Expression (Rule): # 10..12 " 0"
-                              - NumericExpression (Rule): # 10..12 " 0"
-                                  - DecimalLiteral (Token): "0" # 11..12
-                      - Comma (Token): "," # 12..13
-                      - NamedArgument (Rule): # 13..20 " gas: 1"
-                          - Identifier (Token): "gas" # 14..17
-                          - Colon (Token): ":" # 17..18
-                          - Expression (Rule): # 18..20 " 1"
-                              - NumericExpression (Rule): # 18..20 " 1"
-                                  - DecimalLiteral (Token): "1" # 19..20
-                  - CloseBrace (Token): "}" # 20..21
+              - NamedArgumentGroups (Rule): # 3..21 "{value: 0, gas: 1}"
+                  - NamedArgumentGroup (Rule): # 3..21 "{value: 0, gas: 1}"
+                      - OpenBrace (Token): "{" # 3..4
+                      - NamedArguments (Rule): # 4..20 "value: 0, gas: 1"
+                          - NamedArgument (Rule): # 4..12 "value: 0"
+                              - Identifier (Token): "value" # 4..9
+                              - Colon (Token): ":" # 9..10
+                              - Expression (Rule): # 10..12 " 0"
+                                  - DecimalNumberExpression (Rule): # 10..12 " 0"
+                                      - DecimalLiteral (Token): "0" # 11..12
+                          - Comma (Token): "," # 12..13
+                          - NamedArgument (Rule): # 13..20 " gas: 1"
+                              - Identifier (Token): "gas" # 14..17
+                              - Colon (Token): ":" # 17..18
+                              - Expression (Rule): # 18..20 " 1"
+                                  - DecimalNumberExpression (Rule): # 18..20 " 1"
+                                      - DecimalLiteral (Token): "1" # 19..20
+                      - CloseBrace (Token): "}" # 20..21
           - ArgumentsDeclaration (Rule): # 21..25 '("")'
-              - OpenParen (Token): "(" # 21..22
-              - PositionalArgumentsList (Rule): # 22..24 '""'
-                  - Expression (Rule): # 22..24 '""'
-                      - AsciiStringLiteralsList (Rule): # 22..24 '""'
-                          - AsciiStringLiteral (Token): '""' # 22..24
-              - CloseParen (Token): ")" # 24..25
+              - PositionalArgumentsDeclaration (Rule): # 21..25 '("")'
+                  - OpenParen (Token): "(" # 21..22
+                  - PositionalArguments (Rule): # 22..24 '""'
+                      - Expression (Rule): # 22..24 '""'
+                          - StringExpression (Rule): # 22..24 '""'
```

Comment:
`StringExpression` here is defined as an `Enum()`, which should not produce its own `NonTerminalKind`, as it will always result in a unique single node from a variant. I wonder if this is expected.

Reply:
I didn't know that we want to inline the enums. I did that in a bit more general fashion in https://github.com/NomicFoundation/slang/pull/650/commits/37f880e6006a43779ad98bfdff28dfc521537fab